File: java/ucotron-sdk/src/main/java/com/ucotron/sdk/types/IngestionMetrics.java
package com.ucotron.sdk.types;

/**
 * Metrics from the ingestion pipeline.
 */
public class IngestionMetrics {
    private int chunks_processed;
    private int entities_extracted;
    private int relations_extracted;
    private int contradictions_detected;
    private long total_us;

    public int getChunksProcessed() { return chunks_processed; }
    public int getEntitiesExtracted() { return entities_extracted; }
    public int getRelationsExtracted() { return relations_extracted; }
    public int getContradictionsDetected() { return contradictions_detected; }
    public long getTotalUs() { return total_us; }
}

File: php/src/Types/IngestionMetrics.php
<?php

declare(strict_types=1);

namespace Ucotron\Sdk\Types;

class IngestionMetrics
{
    public function __construct(
        public readonly int $chunks_created,
        public readonly int $entities_extracted,
        public readonly int $relations_extracted,
        public readonly float $processing_time_ms,
    ) {}

    public static function fromArray(array $data): self
    {
        return new self(
            chunks_created: (int) ($data['chunks_created'] ?? 0),
            entities_extracted: (int) ($data['entities_extracted'] ?? 0),
            relations_extracted: (int) ($data['relations_extracted'] ?? 0),
            processing_time_ms: (float) ($data['processing_time_ms'] ?? 0.0),
        );
    }
}
